mw-webapp/src/utils/DateUtils.ts

```diff
@@ -47,22 +48,28 @@ export class DateUtils {
   /**
    * Get dates between two dates
    */
-  public static getDatesBetween(startDate: Date, endDate: Date): Date[] {
-    const timeDifferenceInMilliseconds = Math.abs(endDate.getTime() - startDate.getTime());
-    const daysAmountInRange = Math.ceil(timeDifferenceInMilliseconds / DAY_MILLISECONDS);
-    const datesList = new Array(daysAmountInRange)
-      .fill(null)
-      .map((item, i) => {
-
-        /**
-         * TODO #573: Correction need because method getShortISODateValue return time in GMT format but we need UTC format
-         */
-        const correction = DAY_MILLISECONDS;
-
-        return new Date(startDate.getTime() + (i * DAY_MILLISECONDS) + correction);
-      });
-
-    return datesList;
+  public static getDatesBetween(startDate: Date, endDate: Date): string[] {
+    const dates: string[] = [];
+
+    /**
+     *GetISODate
+     */
+    const getISODate = (date: Date) => date.toISOString().split("T")[0];
+
+    const isoStartDate = getISODate(startDate);
+    const isoEndDate = getISODate(endDate);
+
+    const currentDate = new Date(isoStartDate);
+    const lastDate = new Date(isoEndDate);
+
+    lastDate.setDate(lastDate.getDate() + ONE_DAY);
+
+    while (currentDate < lastDate) {
+      dates.push(getISODate(currentDate));
+      currentDate.setDate(currentDate.getDate() + ONE_DAY);
+    }
```

Comment:
Lets  implement this logic with some array methods like forEach or map etc.? Let's avoid this kind of loops, it's will be hard to maintain

Reply:
fixed